src/server/cmd/worker/main.go

```diff
@@ -94,11 +94,15 @@ func do(appEnvObj interface{}) error {
 		return fmt.Errorf("error constructing etcdClient: %v", err)
 	}
 
-	// Construct worker API server.
 	pipelineInfo, err := getPipelineInfo(etcdClient, appEnv)
 	if err != nil {
 		return fmt.Errorf("error getting pipelineInfo: %v", err)
 	}
+
+	// Set the auth token that will be used for this client
+	pachClient.SetAuthToken(pipelineInfo.Capability)
```

Comment:
Per our conversation, the way the code is written now makes the most sense. All of the functions in `src/client/pfs.go` and `src/client/pps.go` don't take a context, so either we have to change all of those functions and all of their callers (which includes not just Pachyderm, but also external users of the Pachyderm client library), or we add `SetAuthToken` to `client.APIClient` to make those heavily-used functions continue to work with Pachyderm clusters that have the authorization system enabled.

Also, while pfs and pps make calls to the auth API and the object store, the only place where pfs and pps call each other (that I recall) is that `pps.CreatePipeline()` calls `pfs.CreateRepo()`. It's possible for pps to use the pfs grpc client directly instead of the convenience function in that context (incidentally, calling `SetAuthToken()` in that context actually won't work, because `client.APIClient` isn't thread safe), so rewriting all users of `client.go` just because `CreatePipeline()` can't use `client/{pfs/pps.go,pps/pps.go}` doesn't make sense.